alpha/alpha.ino

```diff
@@ -8,95 +8,168 @@
 
 bool powerSaveTest = false;
 
+bool SavePower = false;
+int stackSize = 1000;
+unsigned long debounceDelay = 50; // the debounce time; increase if the output flickers
+
 void setup() {
   TS_HAL.begin();
   TS_HAL.ble_init();
-  
+
   // Reduce screen brightness to minimum visibility to reduce power consumption
   TS_HAL.lcd_brightness(12);
 
   TS_HAL.lcd_cursor(40, 0);
   TS_HAL.lcd_printf("ALPHA TEST");
 
-  if(powerSaveTest)
+  if (powerSaveTest)
   {
     TS_HAL.lcd_backlight(false);
     TS_HAL.lcd_sleep(true);
   }
 
   OT_ProtocolV2.begin();
-  
+
+  xTaskCreate(
+    traceTask, /* Task function. */
+    "Trace", /* name of task. */
+    stackSize * 10, /* Stack size of task */
+    NULL, /* parameter of the task */
+    2, /* priority of the task */
+    NULL); /* Task handle to keep track of created task */
+
+  xTaskCreate(
+    UITask, /* Task function. */
+    "UI", /* name of task. */
+    stackSize, /* Stack size of task */
+    NULL, /* parameter of the task */
+    3, /* priority of the task */
+    NULL); /* Task handle to keep track of created task */
 }
 
 int skips = 0;
 
 void loop() {
   TS_HAL.update();
 
-  if(!powerSaveTest)
+  if (!powerSaveTest)
```

Comment:
temporarily move these into ui task

Reply:
ui task will be the primary owner of lcd calls. no other task should access the lcd.